src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java

```diff
@@ -166,43 +161,23 @@ public void afterAll(ExtensionContext context) throws Exception {
 
   // This method sets the scripts for the first time to generate the swarm
   private void setFirstScripts(ExtensionContext context) {
+    HiveShellBuilder hiveShellBuilder = new HiveShellBuilder();
     try {
-      Set<Field> fields = ReflectionUtils.getAllFields(context.getRequiredTestClass(), withAnnotation(HiveSQL.class));
-
       scriptsUnderTest.clear();
-
+      Set<Field> fields = ReflectionUtils.getAllFields(context.getRequiredTestClass(), withAnnotation(HiveSQL.class));
       Preconditions.checkState(fields.size() == 1, "Exactly one field should be annotated with @HiveSQL");
       Field field = fields.iterator().next();
-      List<Path> scriptPaths = new ArrayList<>();
       HiveSQL annotation = field.getAnnotation(HiveSQL.class);
-
-      for (String scriptFilePath : annotation.files()) {
-        Path file = Paths.get(Resources.getResource(scriptFilePath).toURI());
-        assertFileExists(file);
-        scriptPaths.add(file);
-      }
-
-      Charset charset = annotation.encoding().equals("") ? Charset.defaultCharset()
-          : Charset.forName(annotation.encoding());
-
-      int index = 0;
-      for (Path path : scriptPaths) {
-        Preconditions.checkState(Files.exists(path), "File %s does not exist", path);
-        try {
-          String sqlText = new String(Files.readAllBytes(path), charset);
-          Script script = new HiveRunnerScript(index++, path, sqlText);
-          scriptsUnderTest.add(script);
-        } catch (IOException e) {
-          throw new IllegalArgumentException("Failed to load script file '" + path + "'");
-        }
-      }
+      List<Path> scriptPaths = getScriptPaths(annotation);
+      
+      Charset charset = annotation.encoding().equals("") ? Charset.defaultCharset() : Charset.forName(annotation.encoding());
+      
+      scriptsUnderTest = hiveShellBuilder.fromScriptPaths(scriptPaths, charset);
+      
+      
```

Comment:
Niiice this method looks way better 🙌🏻  
Theres just a few empty lines that could be cleaned up